bindings/pyroot/JupyROOT/helpers/handlers.py

```diff
@@ -91,33 +113,39 @@ class Runner(object):
     Asynchronous
     >>> print(r.HasFinished())
     True
+    >>> r.Stop()
     '''
     def __init__(self, function):
         self.function = function
-        self.thread = None
+        self.is_running = False
+        self.argument_queue = queue.Queue()
+        self.poller = Poller(runner_obj=self, name = "JupyROOT Runner Thread")
+        self.poller.start()
+
+    def __del__(self):
+        if self.poller.is_alive():
+            self.Stop()
+        self.poller.join()
 
     def Run(self, argument):
         return self.function(argument)
 
     def AsyncRun(self, argument):
-        self.thread = Thread(target=self.Run, args =(argument,))
-        self.thread.start()
+        self.is_running = True
```

Comment:
Isn't it the poller thread the one that sets `is_running`?

Reply:
this is correct. Thanks.

Reply:
I was thinking that perhaps we do not need `is_running` at all... What was its purpose?